docs/architecture.md

```diff
@@ -1,188 +1,260 @@
+somehow? ConfigMap, should be in the diagram? In the OCI registry?-->
 
-At start time, `policy-server` reads its configuration and downloads
-all the Kubewarden policies. Policies can be downloaded from remote
-endpoints like HTTP(s) servers and container registries.
+When creating, modifying or deleting a `ClusterAdmissionPolicy`,
+a reconciliation loop activates in `kubewarden-controller`,
+for the `PolicyServer` owning the policy.
+This reconciliation loop creates a ConfigMap with all the polices bound to the `PolicyServer`.
+Then the Deployment rollout of the `policy-server` starts.
+It results in starting the new `policy-server` instance with the updated configuration.
 
-Policies' behaviour can be tuned by users via policy-specific configuration
-parameters. Once all the policies are downloaded, `policy-server` will ensure
-the policy settings provided by the user are valid.
+<!--TODO:Next paragraph. Download all the policies, that the configuration
+specifies, right?-->
 
-`policy-server` performs the validation of policies's settings by
-invoking the `validate_setting` function exposed by each policy.
-This topic is covered more in depth inside
-of [this section](/writing-policies/spec/01-intro-spec.md) of the documentation.
+At start time, the `policy-server` reads its configuration
+and downloads all the Kubewarden policies specified.
+You can download policies from remote HTTP servers and container registries.
 
-`policy-server` will exit with an error if one or more policies received wrong
-configuration parameters from the end user.
+You use policy configuration parameters to tune a policies' behavior.
+After startup and policy download the `policy-server`
+checks the policy settings provided by the user are valid.
 
-If all the policies are properly configured, `policy-server` will spawn a
-pool of worker threads to evaluate incoming requests using the Kubewarden
-policies specified by the user.
+The `policy-server` validates policy settings by invoking
+the `validate_setting` function exposed by each policy.
+There is further documentation in the
+[writing policies section](/writing-policies/spec/01-intro-spec.md)
+of the documentation.
 
-Finally, `policy-server` will start a HTTPS server that listens to incoming
-validation requests. The web server is secured using the TLS key and certificate
-that have been previously created by `kubewarden-controller`.
+<!--TODO:Next paragraph. 'end-user'? How are the configuration parameters
+specified for the `policy-server`-->
 
-Each policy is exposed by the web server via a dedicated path that follows this
-naming convention: `/validate/<policy ID>`.
+The `policy-server` exits with an error if one or more policies
+received wrong configuration parameters from the end-user.
 
-This is how the cluster looks like once the initialization of `policy-server`
-is completed:
+<!--TODO:Can we show the pool of worker threads in the diagram? Or maybe
+it's the stacked things?-->
```

Comment:
We can add this in the new version of the diagram if this is important.

Reply:
I'm not sure it adds much, the pool is abstract at this moment, and we don't get into details on the scheduling inside of the pool.

Reply:
I don't think it's important if you guys don't. It's an architecture overview. The idea being so that a Kubewarden User has a grasp of how KW is put together and works to best use its features. So, describing worker threads in any detail is getting in too deep here. Thanks.

Reply:
Thanks.